### ISSUE TITLE
Show marker on map after searching for coordinates

### ISSUE THREAD

**Issue by carstenhag** [openstreetmap/iD#7435]
After searching for a specific location, ID should show a location marker on the specifc location. Currently, it just zooms in somewhat, but it's unclear where exactly the coordinate is from.

My usecase is: I take a photo, then extract the coordinates and search with them on ID, for example `48.304564,10.9553521`. I then want to tag the node I took a photo from at the specific location.

Google Maps shows a location marker after searching with coordinates:

![chrome_d1s8b51cs2](https://user-images.githubusercontent.com/3776997/76705817-20ac6600-66e3-11ea-9193-846b1be9d815.png)

ID, as said, just zooms in and leaves me with a big area to locate the thing I want to tag:

![m4quSr2eql](https://user-images.githubusercontent.com/3776997/76705824-36219000-66e3-11ea-947b-ff5c93e1e64a.jpg)

**Comment by jidanni** [comment]
Not allowed, according to #6830 #4197 #4218 #6173 #5117 #3088 etc. etc. etc.

**Comment by carstenhag** [comment]
Okay, so I read some of these. Clearly I was there because I took the picture. I can tell whether the found location was along my trip or is in China (or in the Atlantic sea, wherever the mixed up coordinates would put me at).

**Comment by carstenhag** [comment]
>That's exactly why we don't show a marker - Because there is no map feature there to edit. iD is an editor, not a search engine, so we only draw markers where things are.

Clearly there is a search field, so the point is moot imo. The search is there for orientation. An editor needs orientation, else I would put amenities in the wrong city? Seems like an excuse to me, imo.

**Comment by jidanni** [comment]
Maybe there could be an independently distributed plugin for this. Alas I [read]( https://wiki.openstreetmap.org/wiki/ID) iD is "usable directly in the browser without plugins"...

**Comment by jidanni** [comment]
It seems this is a perfect candidate for a simple plugin. Plugins were mentioned in #1392 #3836 #1391 .

**Comment by jidanni** [comment]
Today I devised a test:

Does the (frustratingly "invisible/nonexistent" coordinate search result) marker stay at the center of the screen as one
zooms in with the + button, or does it drift to the side?

To do this experiment, we
1. Take the coordinates of a [known point](https://www.openstreetmap.org/node/4778301328),
> 24.1843080, 120.8723690

2. Tinker with them a little (else the search function recognizes the
point, ruining the experiment.)
> 24.184308**1**, 120.872369**1**

3. Put them in the search box, and click on the "Location" provided.

![27316-0](https://user-images.githubusercontent.com/1230959/100610548-e525fa80-334a-11eb-82d6-409cbba44e7b.jpg)

Indeed, no drift observed all the way to the maximum zoom level.

**Comment by jidanni** [comment]
Hack to search for exact coordinate position and actually see where it is (instead of just "somewhere in the center of the screen")!:  https://github.com/openstreetmap/iD/issues/4197#issuecomment-830928223 .

**Comment by tekeroth-snapcode** [comment]
I dont understand these comments and "we dont do this". If someone searches for a specific coordinate, why not show it on the map? It has nothing to do with editing the map, or dragging nodes or anything like it. It is just a question of showing the exact coordinate on the map. 

How is this a problem?

**Comment by 1ec5** [comment]
> It has nothing to do with editing the map, or dragging nodes or anything like it. It is just a question of showing the exact coordinate on the map.

The original post and most of the linked issues have been explicit that this is about editing the map. In any event, you can triangulate your cursor over the desired coordinate interactively, by enabling the Measurement panel and observing the coordinates as you move the cursor around the center of the map. If you need to repeat this process many times for many coordinates that come from something like EXIF data (as described above) or a GPS trace, you can add a GPX or GeoJSON file as a custom data layer using the Map Data panel.

**Comment by tekeroth-snapcode** [comment]
Hello, and thanks for the reply!
The creator of this thread mentioned not a thing about editing the map, if you read the post. He simply says that there is no map marker when searching for a coordinate. And he is right.

It is a basic use case that when you add a coordinate, that that position is shown on the map. Not in edit mode, but in the normal search/view mode. 

The suggestion you have, to "triangulate your cursor", is frankly not a solution. The simple, straightforward solution is of course to show the coordinate with a map marker, just like google does. This is a no-brainer, and I cannot understand why the OSM admins are so against it. It truly baffles me.

**Comment by 1ec5** [comment]
> My usecase is: I take a photo, then extract the coordinates and search with them on ID, for example 48.304564,10.9553521. I then want to tag the node I took a photo from at the specific location.

**Comment by 1ec5** [comment]
> The suggestion you have, to "triangulate your cursor", is frankly not a solution. The simple, straightforward solution is of course to show the coordinate with a map marker, just like google does. This is a no-brainer, and I cannot understand why the OSM admins are so against it. It truly baffles me.

This issue remains open, in case someone comes along contributing a fix. I don’t know who you’re referring to as the OSM admins against this idea, but what I suggested was a workaround, in case the lack of this marker was impeding your edits somehow.

**Comment by carstenhag** [comment]
As the issue creator, maybe I can clarify:

Yes, I want the marker to be shown both on the read-only and the editing parts.

No, I will not blindly use the found location to create a node. I will still use common sense, trying to find an existing node/way. If nothing's there yet, I will use common sense such as using the satellite map or only creating a note with the image.

**Comment by tekeroth-snapcode** [comment]
Alright, he talks about ID, I admit, I dont know what that is, I guess the editor then.

But in any case, the primary missing feature is in the view/search-mode. But I would also guess, it is missing in the edit mode too. Several issues have been on this, it seems, and some ppl (above) says:

> Not allowed, according to https://github.com/openstreetmap/iD/issues/6830 https://github.com/openstreetmap/iD/issues/4197 https://github.com/openstreetmap/iD/issues/4218 https://github.com/openstreetmap/iD/issues/6173 https://github.com/openstreetmap/iD/issues/5117 https://github.com/openstreetmap/iD/issues/3088 etc. etc. etc.

This is in my opinion completely wrong. Of course the map should show a marker where the coordinate is, as you have entered them. Both in read-only and edit mode.

The funnies thing is that OSM shows the coordinates in the "Search Results" on the left, and if you hover the mouse over the coordinate, it shows up on map, but _only_ if you hover 🤣 Oh why oh why? It is almost like an insult ;-)

**Comment by 1ec5** [comment]
This repository is about OpenStreetMap’s Web-based editor, iD. A separate project, [openstreetmap-website](https://github.com/openstreetmap/openstreetmap-website/), is responsible for the “read-only mode”. (iD is actually a frame embedded inside that website.) There are some issues about how the website’s coordinate-based search result doesn’t do anything except on hover: openstreetmap/openstreetmap-website#2197 openstreetmap/openstreetmap-website#3827.

One of the reasons nothing has happened on this front is that there has been a steady stream of strident, at times counterproductive lobbying for a coordinate-based editing tool over the years. Even though that isn’t quite what’s being requested here, the tenor of those requests has previously dissuaded people from contributing to this area of the codebase.

In case anyone is interested in designing a working solution for a transient marker when searching (but not editing), here’s where the sidebar detects that the search terms are a coordinate pair and displays the coordinate pair as a search result:

https://github.com/openstreetmap/iD/blob/5f92841a8ee5bb371a7f3898725493a97aa5adca/modules/ui/feature_list.js#L126-L137

Here’s where the sidebar reacts to the cursor hovering over the search result, highlighting and unhighlighting it:

https://github.com/openstreetmap/iD/blob/5f92841a8ee5bb371a7f3898725493a97aa5adca/modules/ui/feature_list.js#L327-L338

**Comment by tekeroth-snapcode** [comment]
Thank you very much for the input, I appreciate it.

I also found that @jidanni found a workaround:
https://github.com/openstreetmap/openstreetmap-website/issues/2197#issuecomment-830758106

**Comment by jidanni** [comment]
Yes but my workaround for the website makes it even more maddening for iD:
There I am looking at the exact position on the website, but there's no way to somehow transport that marker to iD!!! !! !!!! !
